lib/rubocop/cop/rspec/remove_const.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks that `remove_const` is not used in specs.
+      #
+      # @example
+      #   # bad
+      #   it 'does something' do
+      #     Object.send(:remove_const, :SomeConstant)
+      #   end
+      #
+      #   before do
+      #     SomeClass.send(:remove_const, :SomeConstant)
+      #   end
+      #
+      class RemoveConst < Base
+        include RuboCop::RSpec::Language
+        extend RuboCop::RSpec::Language::NodePattern
+
+        MSG = 'Do not use remove_const in specs. ' \
+              'Consider using e.g. `stub_const`.'
+
```

Comment:
I thought we had a cop for reminding about this, but: You should probably add a `RESTRICT_ON_SEND = %i[send __send__].freeze` to help RuboCop reduce the number of `#on_send` methods it needs to call.

Reply:
pushed that one :)